src/doc/tarpl/meet-safe-and-unsafe.md

```diff
@@ -0,0 +1,98 @@
+% Meet Safe and Unsafe
+
+Programmers in safe "high-level" languages face a fundamental dilemma. On one
+hand, it would be *really* great to just say what you want and not worry about
+how it's done. On the other hand, that can lead to some *really* poor
+performance. It may be necessary to drop down to less clear or idiomatic
+practices to get the performance characteristics you want. Or maybe you just
+throw up your hands in disgust and decide to shell out to an implementation in
+a less sugary-wonderful *unsafe* language.
+
+Worse, when you want to talk directly to the operating system, you *have* to
+talk to an unsafe language: *C*. C is ever-present and unavoidable. It's the
+lingua-franca of the programming world.
+Even other safe languages generally expose C interfaces for the world at large!
+Regardless of *why* you're doing it, as soon as your program starts talking to
+C it stops being safe.
+
+With that said, Rust is *totally* a safe programming language.
+
+Well, Rust *has* a safe programming language. Let's step back a bit.
+
+Rust can be thought of as being composed of two
+programming languages: *Safe* and *Unsafe*. Safe is For Reals Totally Safe.
+Unsafe, unsurprisingly, is *not* For Reals Totally Safe. In fact, Unsafe lets
```

Comment:
Warning: Big Overhaul Suggestion Coming

I would prefer to use words (or phrases) other than just capitalized "Safe" and "Unsafe" for the two language dialects under discussion here.  Even "SafeRust" and "UnsafeRust" would work (and I think you already use "Safe Rust" in at least one point in the text), though perhaps other options would work ("Trust" and "Distrust" ?)

The reason is the words "safe" and "unsafe" are going to occur a lot in this document, and I think relying solely on the  case of the first character (and/or context of the word itself, e.g. when it is the first word in a sentence) to distinguish the two puts an unexpected burden on the reader.